realm/src/androidTest/java/io/realm/RealmJsonTest.java

```diff
@@ -479,6 +493,37 @@ public void testCreateOrUpdateJsonObject() throws JSONException {
         assertEquals("bar", newObj.getColumnString());
     }
 
+    public void testCreateOrUpdateJsonObject_ignoreUnsetProperties() {
+        testRealm.beginTransaction();
+        AllTypesPrimaryKey obj = testRealm.createOrUpdateObjectFromJson(AllTypesPrimaryKey.class, "{ \"columnLong\":1, \"columnString\": \"foo\" }");
+        obj.setColumnBoolean(true);
```

Comment:
If you swap line 498 and 499, you will check if `createOrUpdateObjectFromJson()` does not reset to default value (0/false). Or what am I missing?